### ISSUE TITLE
mark_as_read! not working without timeout

### ISSUE THREAD

**Issue by storuky** [ledermann/unread#52]
Hello. I found a strange bug. When I execute this single-line code inside console:
```ruby
Message.create; Message.mark_as_read! :all, :for => User.first; Message.unread_by(User.first)
```
I have this result, but it must be empty array:
```ruby
 [#<Message:0x007f8cd8893318
  id: 18,
  created_at: Fri, 22 May 2015 23:52:09 MSK +03:00,
  updated_at: Fri, 22 May 2015 23:52:09 MSK +03:00>]
```
When I execute this code with sleep, it works correctly:
```ruby
Message.create; sleep(1); Message.mark_as_read! :all, :for => User.first; Message.unread_by(User.first)
```
Chat - is a realtime application. It should not work so...

**Comment by ledermann** [comment]
Hm, I just pushed a [fix for a memoization](https://github.com/ledermann/unread/commit/0c68d0971d557a2633a36d70cd157def0611f9a8) issue, which can be a fix for your issue, too.

But: Is your code above real code? Using `User.first` all the time should avoid memoization problems in general, so my fix doesn't change anything for your issue. Please: Check again with the master branch. If your issue still exists, it would be helpful to get a failing test.

**Comment by ledermann** [comment]
It seems the real fix was made in #51. Please upgrade to v0.6.1.
